api/src/main/java/org/openmrs/ui/framework/fragment/FragmentFactory.java

```diff
@@ -1,576 +1,600 @@
-            temp += "Expected: " + expected.getName() + " (cl: " + expected.getClassLoader() + ")";
-            temp += " | Actual: ";
-            if (params[i] == null) {
-                temp += "null";
-            } else {
-                temp += params[i].getClass().getName() + " (cl: " + params[i].getClass().getClassLoader() + ")";
-            }
-            lines.add(temp);
-        }
-        return "\n" + OpenmrsUtil.join(lines, "\n");
-    }
-
-    public boolean fragmentExists(String providerName, String fragmentName) {
-		Object controller = getController(providerName, fragmentName);
-		if (controller != null) {
-			return true;
-		}
-		
-		FragmentView view = getView(new FragmentRequest(providerName, fragmentName), fragmentName);
-		return (view != null);
-	}
-	
-	public <T> T convert(Object obs, Class<T> toType) {
-		return conversionService.convert(obs, toType);
-	}
-	
-	public <T> Map<String, T> getSpringBeansOfType(Class<T> ofType) {
-		return applicationContext.getBeansOfType(ofType);
-	}
-	
-	public MessageSource getMessageSource() {
-		return messageSource;
-	}
-	
-	public ExtensionManager getExtensionManager() {
-		return extensionManager;
-	}
-	
-	public ConversionService getConversionService() {
-		return conversionService;
-	}
-
-    public FormatterService getFormatterService() {
-        return formatterService;
-    }
-
-    public ServletContext getServletContext() {
-		return servletContext;
-	}
-
-    public void setModelConfigurators(List<FragmentModelConfigurator> modelConfigurators) {
-        this.modelConfigurators = modelConfigurators;
-    }
-
-    public List<FragmentModelConfigurator> getModelConfigurators() {
-        return modelConfigurators;
-    }
-
-    public void setPossibleFragmentControllerArgumentProviders(List<PossibleFragmentControllerArgumentProvider> possibleFragmentControllerArgumentProviders) {
-        this.possibleFragmentControllerArgumentProviders = possibleFragmentControllerArgumentProviders;
-    }
-
-    public void setPossibleFragmentActionArgumentProviders(List<PossibleFragmentActionArgumentProvider> possibleFragmentActionArgumentProviders) {
-        this.possibleFragmentActionArgumentProviders = possibleFragmentActionArgumentProviders;
-    }
-
-    public void setSessionFactory(SessionFactory sessionFactory) {
-        this.sessionFactory = sessionFactory;
-    }
-}
+package org.openmrs.ui.framework.fragment;
+
+import org.openmrs.api.APIAuthenticationException;
+import org.openmrs.ui.framework.RequestValidationException;
+import org.openmrs.ui.framework.UiFrameworkException;
+import org.openmrs.ui.framework.UiFrameworkUtil;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.extension.ExtensionManager;
+import org.openmrs.ui.framework.formatter.FormatterService;
+import org.openmrs.ui.framework.fragment.action.FailureResult;
+import org.openmrs.ui.framework.interceptor.FragmentActionInterceptor;
+import org.openmrs.ui.framework.page.*;
```

Comment:
We don't like wildcard imports in OpenMRS (except in test classes). Can you please change this IntelliJ setting?